hazelcast-client-new/src/main/java/com/hazelcast/client/spi/impl/ClientInvocation.java

```diff
@@ -196,7 +197,7 @@ public void notifyException(Throwable exception) {
             return;
         }
 
-        if (exception instanceof IOException || exception instanceof HazelcastInstanceNotActiveException) {
+        if (exception instanceof IOException || exception instanceof HazelcastInstanceNotActiveException || exception instanceof RetryableAuthenticationException) {
```

Comment:
I though on the issue a little bit more. I think we can safely make all AuthenticationExceptions retryable by changging 
exception instanceof RetryableAuthenticationException 
to 
exception instanceof AuthenticationException.
And remove the new RetryableAuthenticationException.

Since all authentications must be done over a connection and this request will be bind to single connection, AuthenticationRequests will never be retried. If any other invocation get this AuthenticationException, it can safely retry.